gtk/gtk.go

```diff
@@ -9506,6 +9522,130 @@ func (v *TreeRowReference) Valid() bool {
 	return gobool(c)
 }
 
+/*
+ * GtkTreeModelSort
+ */
+
+// TreeModelSort is a representation of GTK's GtkTreeModelSort
+type TreeModelSort struct {
+	*glib.Object
+
+	// Interfaces
+	TreeModel
+}
+
+// native returns a pointer to the underlying GtkTreeModelSort
+func (v *TreeModelSort) native() *C.GtkTreeModelSort {
+	if v == nil || v.GObject == nil {
+		return nil
+	}
+	p := unsafe.Pointer(v.GObject)
+	return C.toGtkTreeModelSort(p)
+}
+
+func marshalTreeModelSort(p uintptr) (interface{}, error) {
+	c := C.g_value_get_object((*C.GValue)(unsafe.Pointer(p)))
+	obj := glib.Take(unsafe.Pointer(c))
+	return wrapTreeModelSort(obj), nil
+}
+
+func wrapTreeModelSort(obj *glib.Object) *TreeModelSort {
+	tm := wrapTreeModel(obj)
+	return &TreeModelSort{obj, *tm}
+}
+
+func (v *TreeModelSort) toTreeModel() *C.GtkTreeModel {
+	if v == nil {
+		return nil
+	}
+	return C.toGtkTreeModel(unsafe.Pointer(v.GObject))
+}
+
+func (v *TreeModelSort) toTreeSortable() *C.GtkTreeSortable {
+	if v == nil {
+		return nil
+	}
+	return C.toGtkTreeSortable(unsafe.Pointer(v.GObject))
+}
+
+// TreeModelSortNew is a wrapper around gtk_tree_model_sort_new_with_model():
+func TreeModelSortNew(model ITreeModel) (*TreeModelSort, error) {
+	var mptr *C.GtkTreeModel
+	if model != nil {
+		mptr = model.toTreeModel()
+	}
+	c := C.gtk_tree_model_sort_new_with_model(mptr)
+	if c == nil {
+		return nil, nilPtrErr
+	}
+
+	tms := wrapTreeModelSort(glib.Take(unsafe.Pointer(c)))
+	return tms, nil
+}
+
+// GetModel is a wrapper around gtk_tree_model_sort_get_model()
+func (v *TreeModelSort) GetModel() ITreeModel {
+	c := C.gtk_tree_model_sort_get_model(v.native())
+	if c == nil {
+		return nil
+	}
+	m := wrapTreeModel(glib.Take(unsafe.Pointer(c)))
+	return m
+}
+
+// ConvertChildPathToPath is a wrapper around gtk_tree_model_sort_convert_child_path_to_path().
+func (v *TreeModelSort) ConvertChildPathToPath(childPath *TreePath) *TreePath {
+	path := C.gtk_tree_model_sort_convert_child_path_to_path(v.native(), childPath.native())
+	if path == nil {
+		return nil
+	}
+	p := &TreePath{path}
+	return p
+}
+
+// ConvertChildIterToIter is a wrapper around gtk_tree_model_sort_convert_child_iter_to_iter().
+func (v *TreeModelSort) ConvertChildIterToIter(sortIter, childIter *TreeIter) bool {
+	return gobool(C.gtk_tree_model_sort_convert_child_iter_to_iter(v.native(), sortIter.native(), childIter.native()))
+}
+
+// ConvertPathToChildPath is a wrapper around gtk_tree_model_sort_convert_path_to_child_path().
+func (v *TreeModelSort) ConvertPathToChildPath(sortPath *TreePath) *TreePath {
+	path := C.gtk_tree_model_sort_convert_path_to_child_path(v.native(), sortPath.native())
+	p := &TreePath{path}
+	return p
+}
+
+// ConvertIterToChildIter is a wrapper around gtk_tree_model_sort_convert_iter_to_child_iter().
+func (v *TreeModelSort) ConvertIterToChildIter(childIter, sortIter *TreeIter) {
+	C.gtk_tree_model_sort_convert_iter_to_child_iter(v.native(), childIter.native(), sortIter.native())
+}
```

Comment:
same as above:

ConvertIterToChildIter() should have only one input parameter and two return values (in the spirit of golang).

Furthermore, childIter may be nil (even if its [documentation](https://developer.gnome.org/gtk3/unstable/GtkTreeModelSort.html#gtk-tree-model-sort-convert-iter-to-child-iter) does not state it directly: `child_iter An uninitialized GtkTreeIter.`) and will therefore crash when you call it

see `func (v *TreeModelFilter) ConvertIterToChildIter(filterIter *TreeIter)`

Reply:
See above comment for nil values in TreeIter, they are fine but using nil will not make sense and GTK will complain.

Reply:
> > still not 100% sure when to use runtime.SetFinalizer()
> 
> When there is something to free/unref/cleanup.
> Any of these criteria:
> 
>     * it is a `glib.Object` of some sort
> 
>     * it is a C-style array (e.g. `char *`)
> 
>     * it has a custom `free()` function (e.g. `TreePath` or `glib.List`)
> 
> 
> TreeIter is just a plain struct, and similarly a lot of types like GdkWindowEdge are enums, i.e. integers - these don't need a finalizer. Those are taken care of automatically, should they ever escape into the heap memory.

Thanks for the explanation, now this makes total sense to me :)